tests/functional_tests/iterator_tests/test_builtins.py

```diff
@@ -134,11 +156,29 @@ def test_arithmetic_and_logical_builtins(program_processor, builtin, inputs, exp
         assert np.allclose(np.asarray(out), expected)
 
 
```

Comment:
Let's use something like this as test setup

```
@pytest.mark.parametrize(
    "input_value, dtype, expected_value",
    [
        (float64("0.1"), float32, float64(float32("0.1"))),
        (int64(42), bool, int64(1)),
        (int64(2147483648), int32, int64(int32(2147483648))),
    ],
)
@pytest.mark.parametrize("as_column", [False, True])
def test_cast(program_processor, as_column, input_value, dtype, expected_value):
    program_processor, validate = program_processor
    column_axis = IDim if as_column else None

    inp = asfield(*asarray(input_value))[0]
    out = asfield((np.zeros_like(*asarray(expected_value))))[0]

    @fundef
    def sten_cast(value):
        return cast_(deref(value), dtype)

    run_processor(
        sten_cast[{IDim: range(1)}],
        program_processor,
        inp,
        out=out,
        offset_provider={},
        column_axis=column_axis,
    )

    if validate:
        assert out[0] == expected_value
```

Reply:
Let's chat about that in detail. (It works if `as_column == False`, but fails for `True`)